components/VM_Arm/src/main.c

```diff
@@ -715,6 +717,40 @@ static int route_irqs(vm_vcpu_t *vcpu, irq_server_t *irq_server)
     return 0;
 }
 
+/* Different from load_image which loads the image into the guest's address space. This pulls the
+ * image from the FileServer 1MB at a time and stores it into a local buffer
+ */
+static size_t get_file(const char *name, char *buf, size_t max_length)
+{
+    int fd;
+    size_t len = 0;
+
+    if (max_length <= 0) {
```

Comment:
Let's drop  this check. It's always greater or equal to zero. Passing 0 runs into the the loop ad breaks there, so that handled well also

Reply:
Done.